evaporate.js

```diff
@@ -885,7 +886,7 @@
                 }
                 var completedAt = new Date(u.completedAt || FAR_FUTURE);
 
-                return con.partSize === u.partSize && completedAt > HOURS_AGO;
+                return con.partSize === u.partSize && completedAt > HOURS_AGO && me.name === u.awsKey && con.bucket === u.bucket;
```

Comment:
by location, I mean randomly generated AWS key. My workflow works like this:

* Client asks server for a randomly generated key, associated with a particular user
* Server returns key to client, and marks that that user is the author of that key in the database and the upload is "in progress"
* Client then uses Evaporate to upload that file
* Server checks that the key that is being asked to signed is one that user is authorised to upload to
* Once the upload completes, the client notifies the server, and the server marks the upload as completed in the database, and runs the appropriate process (in our case, it's a video transcode - as an aside, these origin files are deleted once the transcode is complete and the transcode is saved to another bucket)

When the user tries to upload that file a second time, we saw Evaporate sending a completely different AWS key to be signed. As we don't let users overwrite files, the server refused to sign the request. It took a lot of head scratching to figure out why Evaporate wasn't paying attention to the `aws_key` parameter we were passing in, and instead trying to upload it to a different location.

I think that's a more normal workflow, but we can agree to disagree. I'll modify the PR to add an option and a warning.